compiler/aarch64/codegen/ARM64Instruction.hpp

```diff
@@ -46,6 +46,16 @@ inline bool constantIsImm9(int32_t intValue)
    return (-256 <= intValue && intValue < 256);
    }
 
+/*
+ * @brief Answers if the signed integer value can be placed in 12-bit field
+ * @param[in] intValue : signed integer value
+ * @return true if the value can be placed in 12-bit field, false otherwise
+ */
```

Comment:
My understanding of the ADDImm and SUBImm instructions is that the 12-bit immediate field is zero-extended to 32/64 bits (regardless of whether it is shifted or not).  This means that only unsigned (positive) values 0<=x<=4095 are suitable.  The code here allows for negative values which is not correct.

Reply:
You are correct.  The immediate field of these instructions are zero-extended.